src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2297,8 +2297,15 @@ private bool HasStructCircularity(BindingDiagnosticBag diagnostics)
                         // If this is a backing field, report the error on the associated property.
                         var symbol = field.AssociatedSymbol ?? field;
```

Comment:
Consider moving this declaration inside the `else` and check `field is SynthesizedPrimaryConstructorParameterBackingFieldSymbol { ... }` instead.